src/io/flutter/survey/FlutterSurveyNotifications.java

```diff
@@ -0,0 +1,162 @@
+
+  private static long SURVEY_START;
+  private static long SURVEY_END;
+
+  static {
+    final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm");
+    dateFormat.setTimeZone(TimeZone.getTimeZone("PST"));
+    try {
+      SURVEY_START = dateFormat.parse("2019/08/12 9:00").getTime();
+      SURVEY_END = dateFormat.parse("2019/08/24 18:00").getTime();
+    }
+    catch (ParseException e) {
+      // Shouldn't happen.
+    }
+  }
+
+  private static boolean isSurveyOpen() {
+    final long now = System.currentTimeMillis();
+    return now >= SURVEY_START && now <= SURVEY_END;
+  }
+
+  interface FlutterSurveyNotifier {
+    void prompt();
+  }
+
+  public static void init(@NotNull Project project) {
+    project.getMessageBus().connect().subscribe(FileEditorManagerListener.FILE_EDITOR_MANAGER, new FileEditorManagerListener() {
+      @Override
+      public void fileOpened(@NotNull final FileEditorManager source, @NotNull final VirtualFile file) {
+        check(file);
+      }
+
+      @Override
+      public void selectionChanged(@NotNull FileEditorManagerEvent event) {
+        final VirtualFile file = event.getNewFile();
+        if (file != null) {
+          check(file);
+        }
+      }
+
+      private void check(@NotNull VirtualFile file) {
+        if (PubRoot.isPubspec(file) || file.getFileType() == DartFileType.INSTANCE) {
+          new FlutterSurveyNotifications(project).checkForDisplaySurvey();
+        }
+      }
+    });
+  }
+
+  @NotNull final Project myProject;
+
+  FlutterSurveyNotifications(@NotNull Project project) {
+    this.myProject = project;
+  }
+
+  private void checkForDisplaySurvey() {
+    // Limit display to the survey window.
+    if (!isSurveyOpen()) return;
+
+    final PropertiesComponent properties = PropertiesComponent.getInstance();
+
+    // Don't prompt more often than every 40 hours.
+    final long lastCheckedMillis = properties.getOrInitLong(FLUTTER_LAST_SURVEY_CHECK_KEY, 0);
+    if (System.currentTimeMillis() - lastCheckedMillis < CHECK_INTERVAL_IN_MS) return;
+
+    properties.setValue(FLUTTER_LAST_SURVEY_CHECK_KEY, String.valueOf(System.currentTimeMillis()));
+
+    // Or, if the survey has already been taken.
+    if (properties.getBoolean(SURVEY_TAKEN)) return;
+
+    final boolean reportAnalytics = FlutterInitializer.getCanReportAnalytics();
+    final String notificationContents = reportAnalytics ?
+                                        "By clicking on this link you agree to share feature usage along with the survey responses." : null;
+
+    final Notification notification = new Notification(
+      FlutterMessages.FLUTTER_NOTIFICATION_GROUP_ID,
+      FlutterIcons.Flutter,
+      SURVEY_TITLE,
+      null,
+      notificationContents,
+      NotificationType.INFORMATION,
+      null
+    );
+
+    notification.addAction(new AnAction(SURVEY_ACTION_TEXT) {
```

Comment:
Ah, here's where this string is displayed. Perhaps something like `Sure` or `Sure!` (or just something more plain but terser than the current text, like `Take survey`)? That would go well with the `No thanks` below.

Reply:
@jayoung-lee ?

Reply:
Either way is fine :)

Reply:
Trimmed.  Thanks!